### ISSUE TITLE
godep on imagebuilder causes problems with old packages

### ISSUE THREAD

**Issue by surajssd** [openshift/imagebuilder#8]
I have written this script, following README. But with this I get issues while compiling, about `fsouza/go-dockerclient` used from different locations when doing `godep restore` and `godep save` it causes some conflicts with the dependencies used in `imagebuilder`, as `github.com/docker/docker/builder/command` not found, it seems to have been removed from upstream docker.

```go
	docker "github.com/fsouza/go-dockerclient"
...
	"github.com/openshift/imagebuilder/dockerclient"
	deployapi "github.com/openshift/origin/pkg/deploy/api"
)

func buildImagesRemote(buildDirPath string, imageName string) error {
	f, err := os.Open(filepath.Join(buildDirPath, "Dockerfile"))
	if err != nil {
		return err
	}
	defer f.Close()

	client, err := docker.NewClientFromEnv()
	if err != nil {
		logrus.Fatal(err)
	}
	e := dockerclient.NewClientExecutor(client)
	e.Out, e.ErrOut = os.Stdout, os.Stderr
	e.AllowPull = true
	e.Directory = buildDirPath
	e.Tag = imageName
	e.AuthFn = nil // ... pass a function to retrieve authorization info
	e.LogFn = func(format string, args ...interface{}) {
		fmt.Fprintf(e.ErrOut, "--> %s\n", fmt.Sprintf(format, args...))
	}

	buildErr := e.Build(f, map[string]string)
	if err := e.Cleanup(); err != nil {
		fmt.Fprintf(e.ErrOut, "error: Unable to clean up build: %v\n", err)
	}
	return nil
}
```

But when compiling I get errors as:
```bash
+ go build -tags experimental -o kompose ./cli/main/
# github.com/skippbox/kompose/pkg/transformer/kubernetes
pkg/transformer/kubernetes/k8sutils.go:55: cannot use client (type
*"github.com/skippbox/kompose/vendor/github.com/fsouza/go-dockerclient".Client) as type
*"github.com/openshift/imagebuilder/vendor/github.com/fsouza/go-dockerclient".Client in
argument to dockerclient.NewClientExecutor
pkg/transformer/kubernetes/k8sutils.go:65: type map[string]string is not an expression
```

Tried doing `godep restore` and `godep save ./...` but then facing this issue of 

```bash
$ godep save ./...
godep: Package (github.com/docker/docker/builder/command) not found
```

**Comment by smarterclayton** [comment]
In order for you to build, you'll need to resolve your differences from the go-dockerclient.  You can do that by deleting the vendor directory under image builder for go-dockerclient.

**Comment by smarterclayton** [comment]
This seems weird - all of that should be vendor protected.  Either way I've merged the 1.12 vendoring and it should be better now.  Reopen if this continues.
